client/core/core.go

```diff
@@ -1515,8 +1517,20 @@ func (c *Core) ReconfigureWallet(appPW []byte, assetID uint32, cfg map[string]st
 	// Reload the wallet with the new settings.
 	wallet, err := c.loadWallet(dbWallet)
 	if err != nil {
-		return newError(walletErr, "error loading wallet for %d -> %s: %v", assetID, unbip(assetID), err)
+		return newError(walletErr, "error loading wallet for %d -> %s: %v",
+			assetID, unbip(assetID), err)
 	}
+
+	// If provided SetNewPW flag is true update wallet's password
+	if setNewPW {
+		err = c.setWalletPassword(appPW, wallet, newPW)
+		if err != nil {
+			return newError(walletErr,
+				"error setting wallet password for %d -> %s: %v",
+				assetID, unbip(assetID), err)
+		}
+	}
```

Comment:
You *could* just do this based on whether `newPW` is `!= nil`. I don't think you need the boolean, but I guess it doesn't hurt.

Reply:
That's was my first assumption but then @buck54321 mentioned that it's possible to set empty password.

Think of the case when have encrypted wallet & want to change to unencrypted wallet, so change name & set epty password which considered as unencrypted.

Reply:
I see.

Reply:
We could use != nil here as we use []byte and not a string which has a default value.

See my last commit: 5710d0e